src/Rewrite.hs

```diff
@@ -65,18 +65,19 @@ quotedUrls :: MonadIO m => (Text -> m ()) -> Args -> ExceptT Text m (Maybe Text)
 quotedUrls log (Args _ attrPth drvFile drvContents) = do
   lift $ log "[quotedUrls] started"
   homepage <- Nix.getHomepage attrPth
-  if "\"\"" == homepage
+  -- Bit of a hack, but the homepage that comes out of nix-env is *always*
+  -- quoted by the nix eval, so we drop the first and last characters.
+  let stripped = T.init . T.tail $ homepage
+  if "\"\"" == homepage || (not $ T.isInfixOf stripped drvContents)
```

Comment:
The whitespace is a little messy with all these ifs, but the RHS here is new and the (cosmetic) bugfix.  See a PR like https://github.com/NixOS/nixpkgs/pull/80860, which claims in the PR description that this rewrite fired, when it didn't actually make a diff.  

In the derivation in question, `homepage = inherit (src) meta.homepage;` , where the `src` package is specified in another file. So since the meta wasn't empty, we ran the Text.replace, and since the string to replace wasn't in the file, we replaced nothing -- but we still claimed to have done something in the PR comment.